### PR TITLE
Improve README.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,14 +9,328 @@ It provides:
 * Type options/flags that parse arguments: `myapp --int-flag=49 enum_rest_arg`
 * Automatic help generation
 * Command aliases
+* Functionality to cache data between runs
+* Functionality to store configurations
+* A UI class to support output at different verbosity levels and different output formats.
```
Comment (kasperl) [inline README.md:14]: Did you intend to terminate the last bullet with dot?

```suggestion
* A UI class to support output at different verbosity levels and different output formats
```

File: README.md [hunk 2]
```diff
@@ -9,14 +9,328 @@ It provides:
...
+- `cli.OptionString`: An option that takes a string value. The `cli.Option` constructor
+  is an alias for this class.
+- `cli.Flag`: An option that takes a boolean value. Flags are treated specially, in
+  that they can be negated with a `no-` prefix. For example, if a flag is called
+  `some-flag`, it can be negated with `--no-some-flag`.
+- `cli.OptionEnum`: An option that takes a value from a set of allowed values.
+
+Users are encouraged to extend the `cli.Option` class and create their own typed options.
+
+## App
```
Comment (kasperl) [inline README.md:120]: Reading it here makes me wonder if it would be better not to abbreviate this.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: README.md (hunk 2 above)
Comment (floitsch) [reply README.md:120]: Will do it in a follow-up PR.

**Pull request merged**